# Введение в математический анализ
## Домашнее задание к уроку 1

### Тема 1. Знакомство с математическим анализом

#### Как относятся друг к другу множество и последовательность? (в ответе использовать слова типа: часть, целое, общее, частное, родитель, дочерний субъект и т.д.)

##### Ответ
__Множество__ - набор элементов имеющих объединяющее свойство. <br/>
__Последовательность__ - набор элементов множества, в котором каждому соответствует натуральное число. <br/>
Фактически, понятие множество входит в понятие последовательность.


#### Прочитать высказывания математической логики, построить их отрицания и установить истинность.
$\forall y\in [0;1]: sgn(y) = 1$<br/>
Для любого y в промежутке от нуля до единицы включая крайние точки справидливо, что сигнум этого y равен единице.
_Утверждение неверно. Сигнум в нуле равняется нулю_

$\forall n\in \mathbb{N}>2:\exists x,y,z \in \mathbb{N}: x^n = y^n + z^n$<br/>
Для любого n принадлежащих множеству натуральных чисел и менше двух справедливо, что найдутся такие x,y,z принадлежащие множеству натуральных чисел, для которых справедливо равенство: $x^n = y^n + z^n$

$\forall x\in \mathbb{R}\exists X \in \mathbb{R}:X>x$<br/>
Для любого x принадлежащего множеству вещественных чисел, найдется такой элемент X принадлежащий множеству вещественных чисел, что он будет больше х

$\forall x\in \mathbb{C}\exists y \in \mathbb{C} : x>y\parallel x<y $<br/>
Для любого `x` принадлежащего множеству комплексных чисел найдется такой `y` принадлежащий множеству комплексных чисел что $x > y$

$\forall y \in [0;\frac{\pi }{2}] \exists \varepsilon >0 : \sin y < \sin (y + \varepsilon )$<br/>
Для любого `y` лежащего на промежутке $[0;\frac{\pi }{2}]$ включая крайние точки найдется такой положительный $\varepsilon$, что $\sin y < \sin (y + \varepsilon )$ <br/>
_Утверждение неверно. При $y=\frac{\pi}{2}$ будет $\sin y > \sin (y + \varepsilon )$_

$\forall y \in [0;\pi) \exists \varepsilon >0 : \cos y > \cos(y+ \varepsilon )$<br/>
Для любого `y` лежащего на промежутке $[0;\pi)$ не включая правую точку найдется такой положительный $\varepsilon$, что $\cos y > \cos(y+ \varepsilon )$

$\exists x : x \notin \{\mathbb{N},\mathbb{Z},\mathbb{Q},\mathbb{R},\mathbb{C}\}$<br/>
Существует такой `x`, который не принадлежит ни ко множеству натуральных, ни целых, ни рациональных, ни вещественных, ни комплексных чисел

### Тема 2. Множество
#### Задание 1 
Даны три множества `a`,`b` и `с`. Необходимо выполнить все изученные виды бинарных операций над всеми комбинациями множеств.

##### Ответ
Пересечение множеств: <br/>
$a \cap b \cap c = \{x \mid x \in a \ \& \ x \in b \ \& \ x \in c \}$

Объединение множеств: <br/>
$a \cup b \cup c = \{x \mid x \in a \ || \ x \in b \ || \ x \in c \}$ <br/>
$|a \cup b \cup c| = |a| + |b| + |c| - |a \cap b| - |b \cap c| - |a \cap c| + |a \cap b \cap c|$

Разность: <br/>
$ a \setminus b \setminus c = \{ x \mid x \in a \ \& \ x \notin b \ \& \ x \notin c \}$

Симметрическая разность: <br/>
$ a \bigtriangleup b \bigtriangleup c = \{ x \mid x \in a \bigoplus x \in b \bigoplus x \in c \}$ <br/>
$ |a \bigtriangleup b \bigtriangleup c| = |a| + |b| + |c| - 2|a \cap b| - 2|b \cap c| - 2|a \cap c| + 3|a \cap b \cap c|$


Декартово произведение: <br/>
$ a \times b \times c = |a| \cdot |b| \cdot |c|$

#### Задание 2 
Выполнить задание 1 на языке Python
##### Решение:

In [3]:
a = set([1, 2, 3, 4])
b = set([3, 4, 5, 6])
c = set([6, 7, 2, 4])

Пересечение множеств:

In [4]:
set.union(a, b, c)

{1, 2, 3, 4, 5, 6, 7}

Объединение множеств:

In [6]:
set.intersection(a, b, c)

{4}

Разность:

In [8]:
 a.difference(b).difference(c)

{1}

Симметрическая разность:

In [15]:
set.union(
     a.difference(b).difference(c),
     b.difference(c).difference(a),
     c.difference(a).difference(b)
)

{1, 5, 7}

Декартово произведение:

__Вот тут я не  смог понять__

### Тема 3. Последовательность
#### Задание 3
Даны 4 последовательности: <br/>
$\{a_n\}^\infty_{n=1} = 2^n -n$ <br/>
$\{b_n\}^\infty_{n=2} = \frac{1}{1-n}$ <br/>
$\{c_n\}^\infty_{n=1} = -1^n + \sqrt{2n}$ <br/>
$\{d_n\}^\infty_{n=1} = (-1)^{2n} + \frac{1}{n^2}$ <br/>
Необходимо: <br/>
1. исследовать их на монотонность; 
2. исследовать на ограниченность;
3. найти пятый по счету член.

##### Решение:

__a)__ $\{a_n\}^\infty_{n=1} = 2^n -n$

Для определения монотонности иследуем разность $a_n - a_{n+1}$: <br/>
$a_n - a_{n+1} = 2^n -n - 2^{n+1} +n + 1 = 2^n(1 - 2) +1 = 1 - 2^n < 0, \forall n \in \mathbb{N} => a_n < a_{n+1}$ <br/>
Следовательно, последовательность является __монотонно возрастающей__

Исследуем ограниченность: <br/>
$\lim_{n \to \infty } 2^n -n \to \infty $, значит __последовательность неограниченна__

$a_5 = 2^5 - 5 = 27$

__b)__ $\{b_n\}^\infty_{n=2} = \frac{1}{1-n}$

Для определения монотонности иследуем разность $b_n - b_{n+1}$: <br/>
$b_n - b_{n+1} = \frac{1}{1-n} - \frac{1}{1-n + 1} = \frac{1}{1-n} - \frac{1}{-n} = \frac{-n\ -\ (1-n)}{(1-n)(-n)} = - \frac{1}{n^2 -n} < 0, \forall n \in \mathbb{N} => b_n < b_{n+1}$ <br/>
Следовательно, последовательность является __монотонно возрастающей__

Исследуем ограниченность: <br/>
$\lim_{n \to \infty } \frac{1}{1-n} \to 0$, значит __последовательность ограниченна__

$b_5 = \frac{1}{1-5} = - \frac{1}{4}$

__c)__ $\{c_n\}^\infty_{n=1} = -1^n + \sqrt{2n}$

Для определения монотонности иследуем разность $с_n - с_{n+1}$: <br/>
$c_n - c_{n+1} = -1^n + \sqrt{2n} - (-1)^{n+1} - \sqrt{2(n+1)} = \sqrt{2}(\sqrt{n}-\sqrt{n+1}) < 0 , \forall n \in \mathbb{N} => c_n < c_{n+1}$ <br/>
Следовательно, последовательность является __монотонно возрастающей__

Исследуем ограниченность: <br/>
$\lim_{n \to \infty } -1^n + \sqrt{2n} = \lim_{n \to \infty } \sqrt{2n} -1 \to \infty$ , значит __последовательность неограниченна__

$c_5 = -1 + \sqrt{2\cdot5} = \sqrt{10} -1 \approx 2.162$

__d)__ $\{d_n\}^\infty_{n=1} = (-1)^{2n} + \frac{1}{n^2}$ <br/>

Для определения монотонности иследуем разность $d_n - d_{n+1}$: <br/>
$d_n - d_{n+1} = (-1)^{2n} + \frac1{n^2} - (-1)^{2(n+1)} + \frac1{(n+1)^2} = \frac{1}{n^2} - \frac{1}{(n+1)^2} > 0, \forall n \in \mathbb{N} => d_n > d_{n+1}$ <br/>
Следовательно, последовательность является __монотонно убывающей__

Исследуем ограниченность: <br/>
$\lim_{n \to \infty } (-1)^{2n} + \frac{1}{n^2} = \pm1$, то есть, имеем ограничение и сверху и снизу, значит __последовательность ограниченна__

$d_5 = (-1)^{2\cdot5} + \frac{1}{5^2} = 1 + \frac{1}{25} = 1.04$

#### Задание 4
$a_1 = 128 , a_{n+1} - a_n = 6$ <br/>
найти 12-й член заданной неявно последовательности

##### Решение:
Перед нами последовательность арифметической прогрессии. <br/>
Последовательность вида $a_1, a_1 + b, a_1 + 2b, a_1 + 3b,...$, где $b = a_{n-1} - a_n = 6$ <br/>

Для нахождения n-ного члена арифметической прогрессии воспользуемся формулой:
$$ a_n = a_1 + (n-1)b $$
И получаем $a_{12} = a_1 - 11b = 128 - 66 = 62$

#### Задание 5
На языке Python предложить алгоритм вычисляющий численно предел <br/>
$\lim_{x \to \infty } \frac{n}{\sqrt[n]{n!}}$ <br/>
с точностью $\varepsilon  = 10^{-7}$

In [66]:
import numpy as np
from itertools import count

In [128]:
eps = 10**(-7)

#Функция лимита
lim_func = lambda n : n * np.prod(np.power(np.arange(2,n+1), -1/n))
lim_n_prev = 1

for n in count(start=2, step=1):
    lim_n = lim_func(n)
    if (lim_n - lim_n_prev) <= eps:
        print(lim_n, n)
        break
    else:
        lim_n_prev = lim_n

2.716988423724237 11779


Получается что предел сходится к экспоненте, и точность $\varepsilon$ достигается на $11779$-ом элементе

#### Задание 6 
Предложить оптимизацию алгоритма, полученного в задании 5, ускоряющую его сходимость.